bindings/src/main/java/io/pravega/storage/filesystem/FileSystemChunkStorage.java

```diff
@@ -0,0 +1,294 @@
+import com.google.common.base.Preconditions;
+import io.pravega.segmentstore.storage.chunklayer.BaseChunkStorage;
+import io.pravega.segmentstore.storage.chunklayer.ChunkAlreadyExistsException;
+import io.pravega.segmentstore.storage.chunklayer.ChunkHandle;
+import io.pravega.segmentstore.storage.chunklayer.ChunkInfo;
+import io.pravega.segmentstore.storage.chunklayer.ChunkNotFoundException;
+import io.pravega.segmentstore.storage.chunklayer.ChunkStorage;
+import io.pravega.segmentstore.storage.chunklayer.ChunkStorageException;
+import io.pravega.segmentstore.storage.chunklayer.ConcatArgument;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.PosixFileAttributes;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Set;
+
+/**
+ * {@link ChunkStorage} for file system based storage.
+ *
+ * Each Chunk is represented as a single file on the underlying storage.
+ * The concat operation is implemented as append.
+ */
+
+@Slf4j
+public class FileSystemChunkStorage extends BaseChunkStorage {
+    //region members
+
+    private final FileSystemStorageConfig config;
+
+    //endregion
+
+    //region constructor
+
+    /**
+     * Creates a new instance of the FileSystemChunkStorage class.
+     *
+     * @param config The configuration to use.
+     */
+    public FileSystemChunkStorage(FileSystemStorageConfig config) {
+        this.config = Preconditions.checkNotNull(config, "config");
+    }
+
+    //endregion
+
+    //region capabilities
+
+    @Override
+    public boolean supportsConcat() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsAppend() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsTruncation() {
+        return false;
+    }
+
+    //endregion
+
+    //region
+
+    @VisibleForTesting
+    protected FileChannel getFileChannel(Path path, StandardOpenOption openOption) throws IOException {
+        return FileChannel.open(path, openOption);
+    }
+
+    @VisibleForTesting
+    protected long getFileSize(Path path) throws IOException {
+        return Files.size(path);
+    }
+
+    @Override
+    protected ChunkInfo doGetInfo(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        try {
+            PosixFileAttributes attrs = Files.readAttributes(Paths.get(config.getRoot(), chunkName),
```

Comment:
as for S3 storage, we should have a common method to perform `Paths.get(config.getRoot(), chunkName)`

it can also be a security issue to let **chunkName** be anystring. 
we should have some validation that **at least** blocks:
- null chars
- relative path sequences ("..", "/","\",".")
- anche chunkname cannot be empty

we cannot assume that the caller is a good guy, even if this is our own code

Reply:
fixed the empty name part.
Now also checking that path is not directory, that should handle second case.